packages/utilities/src/utils/importRemote.ts

```diff
@@ -67,6 +68,29 @@ const loadRemote = (
     );
   });
 
+const loadEsmRemote = async (
+  url: RemoteData['url'],
+  scope: ImportRemoteOptions['scope'],
+  bustRemoteEntryCache: ImportRemoteOptions['bustRemoteEntryCache'],
+) => {
+  const timestamp = bustRemoteEntryCache ? `?t=${new Date().getTime()}` : '';
+  const module = await eval(
+    'import(/* webpackIgnore: true */ `${url}${timestamp}`)',
```

Comment:
you should not need to eval this if you have webpack ignore, if its TS converting it to require, then you can add the option to allow dynamic imports

Reply:
timestamp will cause cache bust every time module attempts to load, and will leak memory. Make sure theres a way to check it, looks like you set it to window to cache so i think its ok

Reply:
> you should not need to eval this if you have webpack ignore, if its TS converting it to require, then you can add the option to allow dynamic imports

In the description I dropped a [link to a TS issue](https://github.com/microsoft/TypeScript/issues/43329), basically when you set the module to "commonjs" TSC compiles it to something like this:

`
const a = Promise.resolve().then(() => __importStar(require("/modules/my-module.js")));
`

Which is removing the comment, even if you disable removing comments (which it does by default). Is there another setting to enable it that I couldn't find? We can set it to ESNext or something but I was having a hard time testing it with Next locally, that did seem to work but compiled it to ESM instead of commonjs.

Reply:
> timestamp will cause cache bust every time module attempts to load, and will leak memory. Make sure theres a way to check it, looks like you set it to window to cache so i think its ok

Great point, I did test it locally and when the component mounts and unmounts it does not make another request to the remote entry file.

Reply:
Resolving using "Node16" going forward, this allows us to support commonjs as a default but will respect .mjs files along with package.json type "module" configurations.